Makefile

```diff
@@ -1,23 +1,45 @@
 test:
 	tox
 
-build:
-	rm -f dist/* && python3 -m pip install --upgrade build && python3 -m build .
+.PHONY: build
+build: setup_dependencies
+	rm -rf dist/* && poetry build && ./rename-wheel.sh
+
+install: setup_dependencies
+	poetry install --with dev
+
+.ONESHELL:
+setup_dependencies: check-poetry check-yarn
+	poetry config virtualenvs.create false
+	poetry self add "poetry-dynamic-versioning[plugin]"
+
+.SILENT:
+.PHONY: check-poetry
+check-poetry:
+	command -v poetry >/dev/null 2>&1 || { echo >&2 "Locust requires the poetry binary to be available in this shell to build the Python package.\nSee: https://docs.locust.io/en/stable/developing-locust.html#install-locust-for-development"; exit 1; }
+
+.SILENT:
+.PHONY: check-yarn
+check-yarn:
+	command -v yarn >/dev/null 2>&1 || { echo >&2 "Locust requires the yarn binary to be available in this shell to build the web front-end.\nSee: https://docs.locust.io/en/stable/developing-locust.html#making-changes-to-locust-s-web-ui"; exit 1; }
 
 frontend_build:
 	yarn webui:install && yarn webui:build
 
 release: build
 	twine upload dist/*
 
-build_docs:
+setup_docs_dependencies:
+	poetry install --with docs
+
+build_docs: setup_docs_dependencies
 	sphinx-build -b html docs/ docs/_build/
 
+.SILENT:
+.PHONY: serve_docs
+serve_docs:
```

Comment:
Does this work better than just browsing the files locally? (file://.../docs/_build/index.html)

Reply:
I don't think it necessarily works _better_ - I just like knowing what an HTTP server will do to the files (this ends up being served by RTD)

Things like the logs _can_ be useful to see if anything is messed up (which reading an HTML file itself won't without looking at the browser console or something)

<img width="898" alt="image" src="https://github.com/locustio/locust/assets/82032454/fe67d571-7b6b-4a2e-9cf1-98fb033592ed">

That and it's just a fairly easy way to load the docs (we could even have the default browser pop open `http://localhost` so that it just starts showing you the docs you built).

It's really mostly a suggestion though, I can pap this out to slim down the file 👍

Reply:
I think it make sense the way you did it. Maybe just add a comment about what it adds compared to browsing the files directly?